File: src/main/java/org/example/command/events/EventStoreImpl.java
package org.example.command.events;

import org.example.command.api.EventStore;

import java.util.concurrent.BlockingQueue;
import java.util.concurrent.LinkedBlockingQueue;

public class EventStoreImpl implements EventStore {

    private BlockingQueue<Event> queue = new LinkedBlockingQueue<>();

    public void storeEvent(Event event) {
        queue.add(event);
    }
    public Event pop() throws InterruptedException {
        return queue.take();
    }

}

File: src/main/kotlin/command/impl/EventStoreImpl.kt
package command.impl

import command.api.EventStore
import command.events.Event
import java.util.concurrent.BlockingQueue

class EventStoreImpl(val queue: BlockingQueue<Event>, val allEvents: MutableCollection<Event>): EventStore {
    override fun storeEvent(event: Event) {
        queue.add(event)
        allEvents.add(event)
    }

    override fun pop(): Event =
        queue.take()

    override fun getAllEvents(): List<Event> =
        allEvents.toList()
}